Add VERIFY_AUTH_MANIFEST mailbox command

mlvisaya opened the pull request:
Add a mailbox command to verify an auth manifest without setting it in
DCCM (in contrast with the SET_AUTH_MANIFEST command).

This will be used during firmware update to verify the downloaded
Auth Manifest without setting it.

mlvisaya committed:
Add VERIFY_AUTH_MANIFEST mailbox command

Add a mailbox command to verify an auth manifest without setting it in
DCCM (in contrast with the SET_AUTH_MANIFEST command).

This will be used during firmware update to verify that the downloaded
Auth Manifest is valid before setting it.

swenson commented on api/src/mailbox.rs:
```diff
@@ -1626,6 +1633,9 @@ impl Default for SetAuthManifestReq {
 }
 
+// VERIFY_AUTH_MANIFEST
+pub type VerifyAuthManifestReq = SetAuthManifestReq;
```
Technically the types are slightly different since when they `impl Request` they have to provide the correct `CommandId`.

mlvisaya replied:
got it. i'll separate the struct for VerifyAuthManifestReq

swenson commented on runtime/tests/runtime_integration_tests/test_authorize_and_stash.rs:
```diff
@@ -1179,3 +1180,180 @@ fn test_authorize_from_staging_address_incorrect_digest() {
+        ),
+        Ok(None) => panic!("Expected a response but got None"),
+        Err(_) => {
+            // Expected error due to invalid manifest
```
After this fails, could you verify that the old manifest still works?

mlvisaya replied:
yes, will do.

mlvisaya committed:
Addressed feedback

- Separate the struct for VerifyAuthManifestReq instead of making it the
same for SetAuthManifestReq.
- In the integration test, after invalid manifest verification fails,
try to authorize a soc image to confirm it's still using the old valid
manifest.

swenson reviewed (approved):
LGTM
